### PR TITLE
add libgdal run requirement on libgrpc

### PR DESCRIPTION
**Pull request by akrherz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
*  Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

refs conda-forge/rasterio-feedstock#274

It seems like we need `libgrpc` participating here as a run dependency as we get this via libgdal -> tiledb -> libgoogle-cloud -> libgrpc and need to maintain pinning.

### CODE REVIEW

**Comment by akrherz** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -134,6 +134,7 @@ outputs:
         - geotiff
         - giflib  # [not win]
         - json-c  # [not win]
+        - libgrpc
```
Comment (ocefpaf) [inline recipe/meta.yaml:137]: We need to understand what dependency was bringing this in and either mark the previous build as broken or create a repodatapatch for this.

Comment (akrherz) [reply recipe/meta.yaml:137]: I don't know the best approach when there's a dependency chain like this and the actual conda-forge pinning happens at the end of the chain, in this case downstream of tiledb and in google-cloud-cpp-feedstock repo, AFAIK

Comment (ocefpaf) [reply recipe/meta.yaml:137]: Yep. It is super hard to figure out. I'm not blocking this PR and it should be merge but we need to dig a bit and ask around to avoid hindering a deeper problem or cluttering the gdal recipe with a fix that should be elsewhere.

TL;DR, let's add a comment on top of that line with the issue number, merge this, patch (preferred) or remove the previous build, and keep asking around until we figure this one out.

Comment (akrherz) [reply recipe/meta.yaml:137]: In theory though, practically all the previous builds are under-specified. I want to review the build logs and then will bump this with your requested comment inclusion.

**Comment by akrherz** [comment]
Oye, now I wonder if `tiledb` needs this treatment first.

File: recipe/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipe/meta.yaml:137]: The joys of maintaining gdal. I don't miss it ;-p

**Comment by akrherz** [comment]
This smells like not the right approach, as we would need to be pinning `libgrpc` in tiledb, libgdal, and rasterio for this to "work", but maybe that is how it is supposed to be?

**Comment by xylar** [comment]
I would check with @h-vetinari, who is pretty expert in these things.  Also, see this comment:
https://github.com/conda-forge/protobuf-feedstock/issues/198#issuecomment-1714603007

**Comment by h-vetinari** [comment]
Yeah, the recipe is tiledb, which has a variant that depends on `google-cloud-cpp`, which itself depends on abseil. It's also pretty clear in the log there (c.f. latest build on main):
```
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libcurl.so.4 found in ['libcurl']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libcurl'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libbz2.so.1.0 found in conda-forge::bzip2-1.0.8-h7f98852_4
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/liblz4.so.1 found in conda-forge::lz4-c-1.9.3-h9c3ff4c_1
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libz.so.1 found in conda-forge::libzlib-1.2.13-hd590300_5
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libzstd.so.1 found in conda-forge::zstd-1.5.2-hfc55251_7
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libssl.so.3 found in conda-forge::openssl-3.1.2-hd590300_0
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libcrypto.so.3 found in conda-forge::openssl-3.1.2-hd590300_0
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libgoogle_cloud_cpp_storage.so.2 found in conda-forge::libgoogle-cloud-2.12.0-h840a212_1
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_cord.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_cordz_info.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_cord_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_cordz_functions.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_exponential_biased.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_cordz_handle.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_synchronization.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_stacktrace.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_graphcycles_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_symbolize.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_debugging_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_malloc_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_demangle_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_crc_cord_state.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_crc32c.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_crc_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_crc_cpu_detect.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_str_format_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libgoogle_cloud_cpp_rest_internal.so.2 found in conda-forge::libgoogle-cloud-2.12.0-h840a212_1
   INFO (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libgoogle_cloud_cpp_common.so.2 found in conda-forge::libgoogle-cloud-2.12.0-h840a212_1
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_time.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_strings.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_strings_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_int128.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_civil_time.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_time_zone.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_bad_variant_access.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_base.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_spinlock_wait.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_bad_optional_access.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_throw_delegate.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_raw_logging_internal.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libabsl_log_severity.so.2301.0.0 found in ['libabseil']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libabseil'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (tiledb,lib/libtiledb.so.2.16): Needed DSO lib/libcrc32c.so.1 found in ['libcrc32c']
WARNING (tiledb,lib/libtiledb.so.2.16): .. but ['libcrc32c'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

I would suggest to rebuild tiledb with an abseil-dependence, that should already fix most of the bleeding I think. Possible too to patch existing tiledb builds (with gcs support) to depend on libabseil 230125. Once the former has been added, the migrator should pass by to rebuild it for the new abseil. PS. not sure what's going on with the curl builds there, 7.x and 8.x are ABI-[compatible](https://daniel.haxx.se/blog/2023/03/20/curl-8-0-0-is-here/).

CC @conda-forge/tiledb

**Comment by ocefpaf** [comment]
> It's also pretty clear in the log there (c.f. latest build on main)

Anyone that parses those logs at the top of their head and says "pretty clear" definitely achieved a higher level of consciousness :smile:

**Comment by ocefpaf** [comment]
PS: if that affects a single build of tiled I prefer that we mark it as broken until we figure out a better approach.

**Comment by h-vetinari** [comment]
> Anyone that parses those logs at the top of their head and says "pretty clear" definitely achieved a higher level of consciousness

😜 

My trick is opening the raw log (because the Azure Pipelines Search is broken except for windows jobs), and then search for the string `"WARNING ("`, which pretty much only ever shows up in the conda link check. 

> PS: if that affects a single build of tiled I prefer that we mark it as broken until we figure out a better approach.

It probably affects more tiledb builds (depending on when the gcp variant was added), just that the breakage didn't show up until a new abseil version was released that surfaced the implicit dependence.

**Comment by akrherz** [comment]
Thanks @h-vetinari , does this mean that `libgdal` needs the same dependency too or just fixing it for tiledb will fix it for "upstream" packages of it?

**Comment by h-vetinari** [comment]
> does this mean that `libgdal` needs the same dependency too or just fixing it for tiledb will fix it for "upstream" packages of it?

I think you mean downstream (i.e. packages depending on tiledb)? In any case, that'll depend whether tiledb uses abseil types in its own external C-API or whether they just use abseil internally. In any case rebuilding tiledb is the first step, and then the link check here should give an indication if anything from gdal also depends on abseil.

**Comment by akrherz** [comment]
> In any case rebuilding tiledb is the first step

thanks again, conda-forge/tiledb-feedstock#209 is in.  The reproducer is trivial if you want to see it in the wild

```
$ mamba create -n repro python=3.11 rasterio
$ python -c 'import rasterio'
```

**Comment by h-vetinari** [comment]
> thanks again, [conda-forge/tiledb-feedstock#209](https://github.com/conda-forge/tiledb-feedstock/pull/209) is in

I thought "in" meant merged... 🤔

**Comment by h-vetinari** [comment]
So the tiledb PR is merged now. Ideally this will already solve most problems. 

However, it would be good to add a run-requirement on `libabseil 20230125` for all gcs-enabled tiledb-builds since https://github.com/conda-forge/tiledb-feedstock/pull/184, to avoid the solver picking one of those builds when trying to solve for newer abseil.

**Comment by h-vetinari** [comment]
FYI: I retriggered CI to see if the problem persists (it doesn't 🥳), but that doesn't mean that this PR should be merged -- it should not be necessary as gdal has no explicit dependence of any kind on abseil/grpc etc., only a transitive one through tiledb (and this is confirmed by the link check in the currently running CI as well).

**Comment by akrherz** [comment]
> it should not be necessary as gdal has no explicit dependence of any kind on abseil/grpc etc.

Agreed, thank you for all the help.
